numba_cuda/numba/cuda/cuda_paths.py

```diff
@@ -45,22 +82,127 @@ def _nvvm_lib_dir():
+            ("Conda environment", get_conda_ctk),
+            ("NVIDIA NVCC Wheel", _get_nvrtc_wheel),
+            ("System", _get_nvrtc_system_ctk),
+        ]
+    )
+    return _find_first_valid_lazy(options)
+
+
+def _get_nvvm_wheel():
+    platform_map = {
+        "linux": ("lib64", "libnvvm.so"),
+        "win32": ("bin", "nvvm64_40_0.dll"),
+    }
+
+    for plat, (dso_dir, dso_path) in platform_map.items():
+        if sys.platform.startswith(plat):
+            break
+    else:
+        raise NotImplementedError("Unsupported platform")
+
+    site_paths = [site.getusersitepackages()] + site.getsitepackages()
+
+    for sp in filter(None, site_paths):
+        nvvm_path = Path(sp, "nvidia", "cuda_nvcc", "nvvm", dso_dir, dso_path)
+        if nvvm_path.exists():
+            return str(nvvm_path.parent)
+
+    return None
+
+
+def get_major_cuda_version():
+    # TODO: remove once cuda-python is
+    # a hard dependency
+    from numba.cuda.cudadrv.driver import get_version
+
+    return get_version()[0]
+
+
+def get_nvrtc_dso_path():
+    site_paths = [site.getusersitepackages()] + site.getsitepackages()
+    for sp in site_paths:
+        lib_dir = os.path.join(
+            sp,
+            "nvidia",
+            "cuda_nvrtc",
+            ("lib" if sys.platform.startswith("linux") else "bin")
+            if sp
+            else None,
+        )
+        if lib_dir and os.path.exists(lib_dir):
+            try:
+                major = get_major_cuda_version()
+                if major == 11:
+                    cu_ver = (
+                        "11.2" if sys.platform.startswith("linux") else "112"
+                    )
+                elif major == 12:
+                    cu_ver = "12" if sys.platform.startswith("linux") else "120"
+                else:
+                    raise NotImplementedError(f"CUDA {major} is not supported")
+
+                return os.path.join(
+                    lib_dir,
+                    f"libnvrtc.so.{cu_ver}"
+                    if sys.platform.startswith("linux")
+                    else f"nvrtc64_{cu_ver}_0.dll",
+                )
+            except RuntimeError:
+                continue
+
+
+def _get_nvrtc_wheel():
+    dso_path = get_nvrtc_dso_path()
+    if dso_path:
+        try:
+            result = ctypes.CDLL(dso_path, mode=ctypes.RTLD_GLOBAL)
+        except OSError:
+            pass
+        else:
+            if sys.platform.startswith("win32"):
+                import win32api
+
+                # This absolute path will
+                # always be correct regardless of the package source
+                nvrtc_path = win32api.GetModuleFileNameW(result._handle)
+                dso_dir = os.path.dirname(nvrtc_path)
+                builtins_path = os.path.join(
+                    dso_dir,
+                    [
+                        f
+                        for f in os.listdir(dso_dir)
+                        if re.match("^nvrtc-builtins.*.dll$", f)
```

Comment:
Did you consider using `glob.glob(os.path.join(dso_dir, "nvrtc-builtins*.dll"))` here?

I'd `assert` that there is exactly one match to be sure. (On Linux there is `libnvrtc-builtins.alt.so`, maybe one day it'll pop up on Windows as well?)

Reply:
This PR just asserts that it exists rather than persisting it to a python global that AFAICT isn't used anywhere. That said, `numba-cuda` isn't running windows tests in CI, and there's no job that's testing this situation on windows. 

I had left it in anyways with the goal of `nvmath-python` being able to _delete_ their patch instead of having to do more additional patching for their supported platforms, but without testing it in our CI, its tough to know if it's working or not...

Reply:
The original patch was meant to pre-load all `nvrtc64_120_0.dll`'s dependencies. As long as all needed DLLs are loaded, at JIT time Windows is happy. (Very different behavior from Linux, I know.)